File: backend/app/contracts/matching.py
from typing import Optional, List, Dict, Any
from proofhire.backend.app.schemas import CoreModel


class MatchingResult(CoreModel):
    job_id: int
    candidate_id: int
    score: float
    reasoning: Optional[Dict[str, Any]] = None
    matched_skills: List[str] = []
    missing_skills: List[str] = []


class MatchingRequest(CoreModel):
    job_id: int
    candidate_ids: Optional[List[int]] = None
    limit: int = 10

File: frontend/src/types/matching.js
export const matchingResultShape = {
  job_id: "number",
  candidate_id: "number",
  score: "number",
  reasoning: "object",
  matched_skills: "array",
  missing_skills: "array",
};

export const matchingRequestShape = {
  job_id: "number",
  candidate_ids: "array",
  limit: "number",
};
